Dockerfile.master.template

```diff
@@ -0,0 +1,168 @@
+      libcurl3-gnutls \
+      libgmp10 \
+      libgmpxx4ldbl \
+      libjson-c3 \
+      libmpfr6 \
+      libprotobuf-c1 \
+      libtiff5 \
+      libxml2 \
+      sqlite3 \
+      curl \
+    && apt-get install -y --no-install-recommends \
+      autoconf2.13 \
+      automake \
+      autotools-dev \
+      bison \
+      build-essential \
+      ca-certificates \
+      cmake \
+      git \
+      g++ \
+      libboost-all-dev \
+      libcgal-dev \
+      libcurl4-gnutls-dev \
+      libgmp-dev \
+      libjson-c-dev \
+      libmpfr-dev \
+      libprotobuf-c-dev \
+      libsqlite3-dev \
+      libtiff-dev \
+      libtool \
+      libxml2-dev \
+      make \
+      pkg-config \
+      postgresql-server-dev-12 \
+      protobuf-c-compiler \
+      xsltproc \
+    # sfcgal
+    && mkdir -p /usr/src/sfcgal \
+    && cd /usr/src/sfcgal \
+    && git init \
+    && git remote add origin https://github.com/Oslandia/SFCGAL.git \
+    && git fetch --depth 1 origin :${SFCGAL_GIT_HASH} \
+    && git reset --hard FETCH_HEAD \
+    && mkdir cmake-build \
+    && cd cmake-build \
+    && cmake .. \
+    && make -j$(nproc) \
+    && make install \
+    && rm /usr/local/lib/libSFCGAL.la \
+    && cd / \
+    && rm -fr /usr/src/sfcgal \
+    # proj4
+    && mkdir -p /usr/src/proj \
+    && cd /usr/src/proj \
+    && git init \
+    && git remote add origin https://github.com/OSGeo/PROJ.git \
+    && git fetch --depth 1 origin :${PROJ_GIT_HASH} \
+    && git reset --hard FETCH_HEAD \
+    && ./autogen.sh \
+    && ./configure --disable-static \
+    && make -j$(nproc) \
+    && make install \
+    && cd / \
+    && rm -fr /usr/src/proj \
+    # geos
+    && mkdir -p /usr/src/geos \
+    && cd /usr/src/geos \
+    && git init \
+    && git remote add origin https://github.com/libgeos/geos.git \
+    && git fetch --depth 1 origin :${GEOS_GIT_HASH} \
+    && git reset --hard FETCH_HEAD \
+    && ./autogen.sh \
+    && ./configure --disable-static \
+    && make -j$(nproc) \
+    && make install \
+    && cd / \
+    && rm -fr /usr/src/geos \
+    # gdal
+    && mkdir -p /usr/src/gdal \
+    && cd /usr/src/gdal \
+    && git init \
+    && git remote add origin https://github.com/OSGeo/gdal.git \
+    && git fetch --depth 1 origin :${GDAL_GIT_HASH} \
+    && git reset --hard FETCH_HEAD \
+    && cd gdal \
+    && ./autogen.sh \
+    && ./configure --disable-static \
+    && make -j$(nproc) \
+    && make install \
+    && cd / \
+    && rm -fr /usr/src/gdal \
+    # postgis
+    && mkdir -p /usr/src/postgis \
+    && cd /usr/src/postgis \
+    && git init \
+    && git remote add origin https://git.osgeo.org/gitea/postgis/postgis.git \
+    && git fetch --depth 1 origin :${POSTGIS_GIT_HASH} \
+    && git reset --hard FETCH_HEAD \
+    && ./autogen.sh \
+# configure options taken from:
+# https://anonscm.debian.org/cgit/pkg-grass/postgis.git/tree/debian/rules?h=jessie
+    && ./configure \
+#       --with-gui \
+    && make -j$(nproc) \
```

Comment:
and extra `make check` before the `make install` would be helpful. ( detecting errors )
( see https://postgis.net/docs/postgis_installation.html#idm728 )